homeassistant/components/hausbus/light.py

```diff
@@ -0,0 +1,306 @@
+            channel.set_light_color(
+                event.getBrightnessRed(),
+                event.getBrightnessGreen(),
+                event.getBrightnessBlue(),
+            )
+        if isinstance(data, rgbDimmerStatus):
+            event = cast(rgbDimmerStatus, data)
+            if (
+                event.getBrightnessBlue() > 0
+                or event.getBrightnessGreen() > 0
+                or event.getBrightnessRed() > 0
+            ):
+                channel.set_light_color(
+                    event.getBrightnessRed(),
+                    event.getBrightnessGreen(),
+                    event.getBrightnessBlue(),
+                )
+            else:
+                channel.light_turn_off()
+        # led event handling
+        if isinstance(data, ledEvOn):
+            event = cast(ledEvOn, data)
+            channel.set_light_brightness(event.getBrightness())
+        if isinstance(data, ledStatus):
+            event = cast(ledStatus, data)
+            if event.getBrightness() > 0:
+                channel.set_light_brightness(event.getBrightness())
+            else:
+                channel.light_turn_off()
+        # light off events
+        if isinstance(data, (DimmerEvOff, ledEvOff, rgbDimmerEvOff)):
+            channel.light_turn_off()
+
+    def turn_on(self, **kwargs: Any) -> None:
+        """Turn on action."""
+        raise NotImplementedError
+
+    def turn_off(self, **kwargs: Any) -> None:
+        """Turn off action."""
+        raise NotImplementedError
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on light."""
+        self.turn_on(**kwargs)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off light."""
+        self.turn_off()
+
+    @callback
+    def async_update_callback(self, **kwargs: Any) -> None:
+        """Light state push update."""
+        state_changed = False
+        if ATTR_ON_STATE in kwargs:
+            if self._attr_is_on != kwargs[ATTR_ON_STATE]:
+                self._attr_is_on = kwargs[ATTR_ON_STATE]
+                state_changed = True
+
+        if ATTR_BRIGHTNESS in kwargs:
+            if self._attr_brightness != kwargs[ATTR_BRIGHTNESS]:
+                self._attr_brightness = kwargs[ATTR_BRIGHTNESS]
+                state_changed = True
+
+        if ATTR_HS_COLOR in kwargs:
+            if self._attr_hs_color != kwargs[ATTR_HS_COLOR]:
+                self._attr_hs_color = kwargs[ATTR_HS_COLOR]
+                state_changed = True
+
+        if state_changed:
+            self.schedule_update_ha_state()
+
+
+class HausbusDimmerLight(HausbusLight):
+    """Representation of a Haus-Bus dimmer."""
+
+    def __init__(
+        self,
+        instance_id: int,
+        device: HausbusDevice,
+        channel: ABusFeature,
+    ) -> None:
+        """Set up light."""
+        super().__init__(instance_id, device, channel)
+
+        self._channel = cast(Dimmer, channel)
```

Comment:
Instead of casting here, could we type the `__init__` parameter as `Dimmer`?

Reply:
I did not think that this is possible, but everything still works as expected :+1: